### PR TITLE
Terraform config for load testing environment

### PR DESCRIPTION
**Pull request by vigneshmsft**
## Context

Create a load-testing environment for apply,

## Changes proposed in this pull request

Terraform configuration for load testing environment
Make commands to deploy changes to enviornment from dev cli
Update deployment instructions documentation.

## Link to Trello card

https://trello.com/c/MdjMOXMj/3947-set-up-load-testing-environment

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
looks great, esp the manual deployment tooling, thanks for adding that! some minor comments but otherwise 👍

File: terraform/workspace_variables/loadtest.tfvars [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# PaaS
+paas_app_environment       = "load-test"
+paas_cf_space              = "bat-prod"
```
[the remark this answers is not in the document]
Comment (vigneshmsft) [reply terraform/workspace_variables/loadtest.tfvars:3]: we want this deployed in the prod space, for it to be able to send metrics to the prometheus instance

File: terraform/workspace_variables/loadtest_backend.tfvars [hunk 2]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "s121d01-apply-rg"
+storage_account_name = "s121d01applyterraform"
+container_name       = "load-test-paas-tfstate"
+key                  = "terraform.tfstate"
```
[the remark this answers is not in the document]
Comment (vigneshmsft) [reply terraform/workspace_variables/loadtest_backend.tfvars:4]: the container_name differentiates it,

File: terraform/workspace_variables/loadtest.tfvars [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+paas_redis_service_plan    = "micro-ha-5_x"
+
+# KeyVault
+key_vault_resource_group    = "s121d01-shared-rg"
+key_vault_name              = "s121d01-shared-kv-01"
+key_vault_app_secret_name   = "APPLY-APP-SECRETS-LOADTEST"
+key_vault_infra_secret_name = "BAT-INFRA-SECRETS-QA"
+
+# Network Policy
+prometheus_app = "prometheus-bat"
```
[the remark this answers is not in the document]
Comment (vigneshmsft) [reply terraform/workspace_variables/loadtest.tfvars:19]: the prometheus app name is prometheus-bat

File: docs/manual-deployment.md [hunk 4]
```diff
@@ -1,47 +1,49 @@
...
+4. Click on the "Run workflow" button, this should trigger a run workflow run and deploy the commit sha to the selected environments.
+
+## Instructions using make commands
 
 **NOTE: Before following the steps below you will need to request an elevation of your rights to the 'contributor' role through PIM in the Azure Portal if working on an app hosted in the test or production subscriptions. Guidance on PIM can be found in the [PIM Guide](pim-guide.md) document. PIM is not required in the development subscription.**
 
-1. Launch the Azure DevOps pages at https://dfe-ssp.visualstudio.com/Become-A-Teacher/_build?definitionId=49&_a=summary.
-1. Locate the desired run from that has failed deployment from the list and select it.
-1. Copy the six character hex string at the start of the run name, without the leading '#' as shown in the following diagram, to the clipboard for use later.
-![Manual Deployment - Diagram 1](manual_deployment_dia01.png)
```
[the remark this answers is not in the document]
Comment (vigneshmsft) [reply docs/manual-deployment.md:22]: done

**Review by cwrw — APPROVED** [review]

**Pull request merged**